p4/gateway/gateway.p4

```diff
@@ -106,10 +124,34 @@ control TpsGwIngress(inout headers hdr,
             data_inspect_packet_ipv6;
             NoAction;
         }
-        size = 1024;
+        size = TABLE_SIZE;
         default_action = NoAction();
     }
 
+    action insert_udp_int_for_udp() {
+        hdr.udp_int.len = hdr.udp.len + ((bit<16>)hdr.int_shim.length * BYTES_PER_SHIM * INT_SHIM_HOP_SIZE) + UDP_HDR_BYTES;
+    }
+
+    table insert_udp_int_for_udp_t {
+        actions = {
+            insert_udp_int_for_udp;
+        }
+        size = TABLE_SIZE;
+        default_action = insert_udp_int_for_udp();
+    }
+
+    action insert_udp_int_for_tcp() {
+        hdr.udp_int.len = ((bit<16>)hdr.int_shim.length * BYTES_PER_SHIM * INT_SHIM_HOP_SIZE) + TCP_HDR_BYTES + UDP_HDR_BYTES;
+    }
+
+    table insert_udp_int_for_tcp_t {
```

Comment:
Why is this specific for TCP and not just general for all none UDP traffic?

Reply:
need to add the different header lengths to the udp_int.len value

Reply:
I am however going to remove the tables and just call the action directly from apply